### PR TITLE
[Faucet] Adding new API with old function call for fallback

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Addition of a feature flag to gate the batch api behavior

## Test Plan 

Unit testing to make sure the send UUID can be found.
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-faucet/src/main.rs [hunk 1]
```diff
@@ -92,6 +92,7 @@ async fn main() -> Result<(), anyhow::Error> {
         .route("/gas", post(request_gas))
         .route("/v1/gas", post(batch_request_gas))
         .route("/v1/status", post(request_status))
+        .route("/v1/gas_old", post(request_gas_v1_interface))
```
Comment (amnn) [inline crates/sui-faucet/src/main.rs:95]: The idea of the fallback is that it's controlled by a config flag on `faucet`, so we shouldn't need a new endpoint, instead, based on a config flag, `/v1/gas` would start serving batch requests without actually batching them.  This way, if there's an issue with the new implementation, we can fall back to the old one without having to tell any client to change which endpoitn they are calling.

File: crates/sui-faucet/src/main.rs [hunk 2]
```diff
@@ -265,6 +266,55 @@ async fn request_gas(
...
+                Json(BatchFaucetResponse::from(FaucetError::Internal(
+                    "Input Error.".to_string(),
+                ))),
+            )
+        }
+    };
+    match result {
+        Ok(_) => {
+            info!(uuid =?id, "Request is successfully served");
+            (StatusCode::CREATED, Json(BatchFaucetResponse::from(id)))
```
Comment (amnn) [inline crates/sui-faucet/src/main.rs:306]: The response structure should be exactly the same as for `batch_request_gas`, i.e. it says that the request has been accepted, when in fact it has already successfully served the response.

File: crates/sui-faucet/src/main.rs (hunk 1 above)
Comment (healthydeve) [reply crates/sui-faucet/src/main.rs]: Changed this structure.

**Review by amnn — APPROVED** [review]
Check out the comment about the status code, but otherwise, looks good!

**Pull request merged**